spec/catalog.yaml

```diff
@@ -396,6 +440,126 @@ components:
         location:
           type: string
 
+    Schema:
```

Comment:
I think we might be able to use this to be able to use the same library and achieve all we have discussed.

Basically, at client side (e.g. Spark, a conversion is needed to convert Spark StructType schema to a schema of this shape), and then at server side, a conversion is needed to convert this schema to Arrow schema shape. (this can be provided by this repository as e.g. `lance-catalog-apache-client-utils`)

Then there is enough information to create the dataset using something like

```
import com.lancedb.lance.catalog.client.apache.utils.ArrowUtils
Dataset.create(allocator, location,
            ArrowUtils.toArrowSchema(restSchema), params)
        .close();
```

Thoughts? @yanghua

Reply:
You mean using a string to represent the left type, right?

Reply:
yes